contracts/interchain-token-service/tests/execute.rs

```diff
@@ -51,3 +62,374 @@ fn execute_fails_with_invalid_message() {
+            message_id: message_id.clone(),
+            source_address: source_address.clone(),
+            contract_address: client.address.clone(),
+            payload_hash,
+        },
+    ];
+    let data_hash = get_approve_hash(&env, messages.clone());
+    let proof = generate_proof(&env, data_hash, signers);
+
+    gateway_client.approve_messages(&messages, &proof);
+
+    client.execute(&source_chain, &message_id, &source_address, &payload);
+
+    goldie::assert!(events::fmt_last_emitted_event::<
+        InterchainTransferReceivedEvent,
+    >(&env));
+}
+
+#[test]
+fn deploy_interchain_token_message_execute_succeeds() {
+    let (env, client, gateway_client, _, signers) = setup_env();
+    register_chains(&env, &client);
+
+    let sender = Address::generate(&env);
+    let sender_bytes = sender.clone().to_xdr(&env);
+    let source_chain = client.its_hub_chain_name();
+    let source_address = Address::generate(&env).to_string();
+
+    let token_id = BytesN::from_array(&env, &[1u8; 32]);
+    let token_metadata = TokenMetadata {
+        name: String::from_str(&env, "Test"),
+        symbol: String::from_str(&env, "TEST"),
+        decimal: 18,
+    };
+
+    let msg = HubMessage::ReceiveFromHub {
+        source_chain: String::from_str(&env, HUB_CHAIN),
+        message: Message::DeployInterchainToken(DeployInterchainToken {
+            token_id: token_id.clone(),
+            name: token_metadata.name.clone(),
+            symbol: token_metadata.symbol.clone(),
+            decimals: token_metadata.decimal as u8,
+            minter: Some(sender_bytes),
+        }),
+    };
+    let payload = msg.abi_encode(&env).unwrap();
+    let payload_hash: BytesN<32> = env.crypto().keccak256(&payload).into();
+
+    let message_id = String::from_str(&env, "test");
+
+    let messages = vec![
+        &env,
+        GatewayMessage {
+            source_chain: source_chain.clone(),
+            message_id: message_id.clone(),
+            source_address: source_address.clone(),
+            contract_address: client.address.clone(),
+            payload_hash,
+        },
+    ];
+    let data_hash = get_approve_hash(&env, messages.clone());
+    let proof = generate_proof(&env, data_hash, signers);
+
+    gateway_client.approve_messages(&messages, &proof);
+
+    client.execute(&source_chain, &message_id, &source_address, &payload);
+
+    goldie::assert!(events::fmt_last_emitted_event::<InterchainTokenDeployedEvent>(&env));
+
+    let token = InterchainTokenClient::new(&env, &client.token_address(&token_id));
+
+    assert!(token.is_minter(&sender));
+    assert_eq!(token.name(), token_metadata.name);
+    assert_eq!(token.symbol(), token_metadata.symbol);
+    assert_eq!(token.decimals(), token_metadata.decimal);
+    assert_eq!(
+        client.token_manager_type(&token_id),
+        TokenManagerType::NativeInterchainToken
+    );
+}
+
+#[test]
+#[should_panic(expected = "Error(Contract, #16)")] // ContractError::InvalidTokenMetadata
```

Comment:
Use assert_contract_err? Should avoid testing panics since it's not easy to refactor/maintain

Reply:
Since `execute` doesn't return a Result, but rather uses `panic_with_error` to return the error to the caller, `assert_contract_err` doesn't work here. Is there a better way to test this?

Reply:
makes sense, we'll need to fix the executable interface